src/middle-earth/comparisons.py

```diff
@@ -0,0 +1,103 @@
+from math import sqrt
+from typing import Any
+
+class Vector:
+    """
+    2D vector with an x- and a y-component.
+    Comparisons between vectors are based on their magnitude.
+    """
+
+    def __init__(self, x: float, y: float) -> None:
+        self.x = x
+        self.y = y
+
+    @property
+    def mag(self) -> float:
+        return sqrt(self.x**2 + self.y**2)
```

Comment:
Add `from math import hypot`.
https://docs.python.org/3/library/math.html#math.hypot
```suggestion
        return hypot(self.x, self.y)
```

Reply:
https://github.com/trag1c/mcoding-all-dunders/pull/4/files#r1170419004